tests/unit/test_solvers/test_solution.py

```diff
@@ -371,115 +370,102 @@ def test_plot(self):
+
+        disc = get_discretisation_for_testing()
+        disc.process_model(model)
+        solution = pybamm.ScipySolver().solve(model, np.linspace(0, 1))
+
+        # test save data
+        with pytest.raises(ValueError):
             solution.save_data(f"{test_stub}.pickle")
 
-            data_load = pybamm.load(f"{test_stub}.pickle")
-            np.testing.assert_array_equal(solution.data["c"], data_load["c"])
-            np.testing.assert_array_equal(solution.data["d"], data_load["d"])
+        # set variables first then save
+        solution.update(["c", "d"])
+        with pytest.raises(ValueError, match="pickle"):
+            solution.save_data(to_format="pickle")
+        solution.save_data(f"{test_stub}.pickle")
 
-            # to matlab
-            solution.save_data(f"{test_stub}.mat", to_format="matlab")
-            data_load = loadmat(f"{test_stub}.mat")
-            np.testing.assert_array_equal(solution.data["c"], data_load["c"].flatten())
-            np.testing.assert_array_equal(solution.data["d"], data_load["d"])
-
-            with pytest.raises(ValueError, match="matlab"):
-                solution.save_data(to_format="matlab")
-
-            # to matlab with bad variables name fails
-            solution.update(["c + d"])
-            with pytest.raises(ValueError, match="Invalid character"):
-                solution.save_data(f"{test_stub}.mat", to_format="matlab")
-            # Works if providing alternative name
-            solution.save_data(
-                f"{test_stub}.mat",
-                to_format="matlab",
-                short_names={"c + d": "c_plus_d"},
-            )
-            data_load = loadmat(f"{test_stub}.mat")
-            np.testing.assert_array_equal(solution.data["c + d"], data_load["c_plus_d"])
-
-            # to csv
-            with pytest.raises(
-                ValueError, match="only 0D variables can be saved to csv"
-            ):
-                solution.save_data(f"{test_stub}.csv", to_format="csv")
-            # only save "c" and "2c"
-            solution.save_data(f"{test_stub}.csv", ["c", "2c"], to_format="csv")
-            csv_str = solution.save_data(variables=["c", "2c"], to_format="csv")
-
-            # check string is the same as the file
-            with open(f"{test_stub}.csv") as f:
-                # need to strip \r chars for windows
-                assert csv_str.replace("\r", "") == f.read()
-
-            # read csv
-            df = pd.read_csv(f"{test_stub}.csv")
-            np.testing.assert_allclose(
-                df["c"], solution.data["c"], rtol=1e-7, atol=1e-6
-            )
-            np.testing.assert_allclose(
-                df["2c"], solution.data["2c"], rtol=1e-7, atol=1e-6
-            )
+        data_load = pybamm.load(f"{test_stub}.pickle")
+        np.testing.assert_array_equal(solution.data["c"], data_load["c"])
+        np.testing.assert_array_equal(solution.data["d"], data_load["d"])
```

Comment:
While we're at it, could you please switch to using `np.testing.assert_allclose()`? xref #4488

Reply:
Since he did not make these tests, I would say that this is outside the scope of this PR. Lets just keep this one simple. Switching to other testing methods could require changes to tolerances and such